src/Vega4Wrapper.js

```diff
@@ -0,0 +1,207 @@
+     * @param {object} uri - An object that will be converted into an url string
+     * @param {object} options - passed by the vega loader
+     * @return {Promise} The sanitized url is provided by the 'href' property. We never load file from local file system.
+     */
+    sanitize(uri, options) {
+        var url = this.objToUrl(uri, options);
+        return new Promise(function(accept) {
+            var result = {href: url, loadFile: false};
+            
+            //return
+            accept(result);
+        });
+    }
+
+    /**
+     * Check if host was listed in the allowed domains, normalize it, and get correct protocol
+     * @param {string} host
+     * @returns {Object}
+     */
+    sanitizeHost(host) {
+        // First, map the host
+        host = (this.domainMap && this.domainMap[host]) || host;
+
+        if (this.testHost('https', host)) {
+            return { host: host, protocol: 'https' };
+        } else if (this.testHost('http', host)) {
+            return { host: host, protocol: 'http' };
+        }
+        return undefined;
+    };
+
+    /**
+     * Test host against the list of allowed domains based on the protocol
+     * @param {string} protocol
+     * @param {string} host
+     * @returns {boolean}
+     */
+    testHost(protocol, host) {
+        if (!this.validators[protocol]) {
+            var domains = this.domains[protocol];
+            if (domains) {
+                this.validators[protocol] = makeValidator(domains, protocol === 'https' || protocol === 'http');
+            } else {
+                return false;
+            }
+        }
+        return this.validators[protocol].test(host);
+    };
+    
+    /**
+     * convert a uri object to a url string
+     * @param {object} uri an object consists of type and essential parameters
+     * @param {object} options used to attach CORS infomation
+     * @returns {string} a complete url
+     */
+    objToUrl(uri, options) {
+        var urlParts = {host: uri.host ? uri.host : options.domain},
+            sanitizedHost = this.sanitizeHost(urlParts.host);
+
+        if (!sanitizedHost) {
+            throw new Error('URL hostname is not whitelisted: ' + urlParts.host);
+        }
+        urlParts.host = sanitizedHost.host;
+
+        switch(uri.type) {
+            case 'tabular':
+            case 'map':
+                // { type: 'tabular', path: 'Data.tab' [, lang: 'en'] }
+                // { type: 'map', path: 'Data.map' [, lang: 'en'] }
+                // Get content of a wiki page belonging to Data namespace, where the path 
+                // is the title of the page with an additional leading '/' which gets 
+                // removed. Uses mediawiki api, and extract the content after the request
+                // Query value must be a valid MediaWiki title string, so we ensure there
+                // is no pipe symbol and ends with .tab or .map
+                if (!/^[^|]+\.(tab|map)$/.test(uri.path)) {
```

Comment:
```suggestion
                if (!/^[^|\x1F]+\.(tab|map)$/.test(uri.path)) {
```

(please look at https://gerrit.wikimedia.org/r/c/mediawiki/extensions/Graph/+/508872 -- we need to apply these changes here too)